tsl/src/compression/compression.h

```diff
@@ -114,6 +194,63 @@ typedef struct CompressionStats
 	int64 rowcnt_post_compression;
 } CompressionStats;
 
+typedef struct PerColumn
+{
+	/* the compressor to use for regular columns, NULL for segmenters */
+	Compressor *compressor;
+	/*
+	 * Information on the metadata we'll store for this column (currently only min/max).
+	 * Only used for order-by columns right now, will be {-1, NULL} for others.
+	 */
+	int16 min_metadata_attr_offset;
+	int16 max_metadata_attr_offset;
+	SegmentMetaMinMaxBuilder *min_max_metadata_builder;
+
+	/* segment info; only used if compressor is NULL */
+	SegmentInfo *segment_info;
+	int16 segmentby_column_index;
+} PerColumn;
+
+typedef struct RowCompressor
+{
+	/* memory context reset per-row is stored */
+	MemoryContext per_row_ctx;
+
+	/* the table we're writing the compressed data to */
+	Relation compressed_table;
+	BulkInsertState bistate;
+	/* segment by index Oid if any */
+	Oid index_oid;
+
+	/* in theory we could have more input columns than outputted ones, so we
+	   store the number of inputs/compressors separately */
+	int n_input_columns;
+
+	/* info about each column */
+	struct PerColumn *per_column;
+
+	/* the order of columns in the compressed data need not match the order in the
+	 * uncompressed. This array maps each attribute offset in the uncompressed
+	 * data to the corresponding one in the compressed
+	 */
+	int16 *uncompressed_col_to_compressed_col;
+	int16 count_metadata_column_offset;
+	int16 sequence_num_metadata_column_offset;
+
+	/* the number of uncompressed rows compressed into the current compressed row */
+	uint32 rows_compressed_into_current_value;
+	/* a unique monotonically increasing (according to order by) id for each compressed row */
+	int32 sequence_num;
+
+	/* cached arrays used to build the HeapTuple */
+	Datum *compressed_values;
+	bool *compressed_is_null;
+	int64 rowcnt_pre_compression;
+	int64 num_compressed_rows;
+	/* if recompressing, we must know this so we can reset the sequence number */
+	bool recompressing;
```

Comment:
If iam not wrong we reset sequence number only for old recompress ( decompress + compress).
May be rename it to is_segmentwise_recompress for better readability.

Reply:
sorry I was a bit confused by this comment, (`we reset sequence number only for old recompress`). If I understand you correctly you mean that this `recompressing` only applies to the case where we are in fact *segmentwise* recompressing? If so, yes you are correct about that, and it makes it clearer if this is renamed as you suggested. I will rename this.